.github/workflows/release-dev.yml

```diff
@@ -5,17 +5,17 @@ on:
     branches: [ main ]
 
 jobs:
-  build_and_test:
+  build:
     uses: ./.github/workflows/build.yml
     with:
-      storeArtifacts: true
-      prodRelease: false
+      store_artifacts: true
+      prod_release: false
     secrets: inherit
 
-  dev_upload_ios:
+  upload_ios:
     name: Upload iOS build to Firebase App Distribution
     runs-on: ubuntu-latest
-    needs: [build_and_test]
```

Comment:
Should we release the app if tests are failing? There might be a slight argument for Dev but I would not accept releasing the app on Prod if tests were failing

Reply:
No the app should not be released if tests are failing. Pull requests should not (and cannot?) be merged into main if tests fail. We will therefore never have a build failing the test suite on main

Reply:
My concern is that this principle is not guarded by pipelines but by a particular configuration we have for our branch protection. I think the contract is more clear when the dependency is expressed in the pipeline

Reply:
I understand your concern but i don't see why you would be able to complete a PR with a failing pipeline to start with? That surely indicates something is wrong

Reply:
The tests are run on the checked out branch. This might be out of sync with the master branch and thus you can't be sure that it will have the same effect on the main branch unless you enforce the constraint that branch is always in sync with main 🙂